src/management-system-v2/app/(dashboard)/[environmentId]/executions/[processId]/page.tsx

```diff
@@ -0,0 +1,283 @@
+    }
+
+    // This is necessary, because bpmn-js throws an error if you try to remove a marker
+    // from an element that doesn't exist
+    flushPreviousStyling();
+  }
+
+  const instances = getVersionInstances(selectedProcess, selectedVersion?.version);
+  const selectedInstance = selectedInstanceId
+    ? instances.find((i) => i.processInstanceId === selectedInstanceId)
+    : undefined;
+
+  let selectedBpmn;
+  if (selectedInstance)
+    selectedBpmn = selectedProcess.versions.find(
+      (v) => v.version === +selectedInstance.processVersion,
+    )!;
+  else if (selectedVersion) selectedBpmn = selectedVersion;
+  else selectedBpmn = getLatestVersion(selectedProcess);
+
+  // When selected coloring changes, this function will change
+  // That in turn will trigger the useEffect inside the BPMNCanvas
+  // If a new instance is selected, the same useEffect will be triggered,
+  // only this time because of the bpmn change
+  // NOTE: selectedColoring is not part of the dependencies to avoid re-rendering
+  // the component on a case where it isn't necessary
+  const applyColoring = useCallback(
+    (coloring?: ColorOptions | ElementLike) => {
+      if (!selectedInstance || !canvasRef.current) return;
+      applyColors(
+        canvasRef.current,
+        selectedInstance,
+        typeof coloring === 'string' ? coloring : selectedColoring,
+      );
+    },
+    [selectedInstance, canvasRef],
+  );
+
+  return (
+    <Content compact wrapperClass={contentStyles.Content}>
+      <div
+        style={{
+          height: '100%',
+        }}
+      >
+        <Toolbar className={styles.Toolbar}>
+          <Space
+            aria-label="general-modeler-toolbar"
+            style={{
+              width: '100%',
+              justifyContent: 'space-between',
+              flexWrap: 'nowrap',
+              alignItems: 'start',
+            }}
+          >
+            <ToolbarGroup>
+              <Select
+                value={
+                  selectedInstanceId && selectedInstance
+                    ? selectedInstance.processInstanceId
+                    : undefined
+                }
+                onSelect={(value) => selectNewBpmn('instance', value)}
+                options={instances.map((instance) => ({
+                  value: instance.processInstanceId,
+                  label: instance.label,
+                }))}
+                placeholder="Select an instance"
+              />
+              <Tooltip title="Start new instance">
+                {/** TODO: implement start new instance */}
+                <Button icon={<PlusOutlined />} />
+              </Tooltip>
+
+              <Tooltip title="Filter by version">
+                <Dropdown
+                  menu={{
+                    items: [
+                      {
+                        key: '-1',
+                        label: selectedVersion ? (
+                          <Typography.Text disabled>none</Typography.Text>
```

Comment:
disabled is maybe a bit too strong here since it will change the mouse to something that pretty clearly indicates that this should not be clicked even though you need to click it to remove the current filter